addon/components/structure-plugin/_private/structure.gts

```diff
@@ -148,8 +154,21 @@ export default class Structure extends Component<Sig> {
     }
   }
 
-  onTitleUpdate = (content: string) => {
-    this.args.updateAttribute('title', content);
+  onTitleUpdate = (content: string, tr: Transaction) => {
+    let addToHistory = false;
+    // every character typed would normally trigger a history event
+    // in normal editor operation, this is ok, as the history plugin is smart
+    // enough to group adjacent edits made in a short timespan (default 500ms)
+    //
+    // but since we trigger an attribute update, the history plugin can no
+    // longer recognize these edits as adjacent, so each character triggers
+    // a history event. This makes undoing title edits tedious, so we mimic the
+    // grouping here using the transaction timestamp
+    if (tr.time - this.historyTimeStamp > 500) {
+      addToHistory = true;
+      this.historyTimeStamp = tr.time;
+    }
+    this.args.updateAttribute('title', content, !addToHistory);
```

Comment:
It doesn't really matter here, as these are attribute updates, but I'd think using the `composition` meta here is more correct than the `addToHistory` meta. `addToHistory: false` transactions can never be undoed, if this transaction here would (at one instance) contain another document change, this change would never be able to be undoed...

Reply:
Idk about "more correct", "composition" is a private and undocumented api after all
and given we check the timestamps, the most history you can lose is 500ms worth of edits

Reply:
Agreed that it's not ideal as it indeed seems to be a private API. 
What I mainly mean with 'incorrect' is the following (in case of real document changes):

https://github.com/user-attachments/assets/660d528f-d19f-44f8-928f-e93ace6084c8

Plugins which react with document changes are in this case also not undoed, this might sometimes be a bit hard to debug.
So in this example it's not really a problem. I just think we should be a bit careful around using the `addToHistory` meta.